packages/rps/src/redux/game-v2/__tests__/player-a.test.ts

```diff
@@ -0,0 +1,214 @@
+import { expectSaga } from 'redux-saga-test-plan';
+import { combineReducers } from 'redux';
+import { gameReducer } from '../reducer';
+import { gameSaga } from '../saga';
+import { localStatesA, bName, bAddress, stake, channelStates, aWeapon, salt } from './scenarios';
+import { joinOpenGame, chooseWeapon, updateChannelState, playAgain, resign } from '../actions';
+import { ChannelState } from '../../../core';
+import * as match from 'redux-saga-test-plan/matchers';
+import { RPSChannelClient } from '../../../utils/rps-channel-client';
+import { randomHex } from '../../../utils/randomHex';
+import { rpsChannelClientMocks } from './helpers';
+
+// need to get the same shape, so that selecting state in the saga works
+const reducer = combineReducers({
+  game: gameReducer,
+});
+
+const client = new RPSChannelClient();
+const { callCreateChannel, callUpdateChannel, callCloseChannel } = rpsChannelClientMocks(client);
+
+const gameState = (localState, channelState?: ChannelState) => ({
+  game: {
+    localState,
+    channelState: channelState || null,
+  },
+});
+
+describe('when joining an open game', () => {
+  it('calls createChannel and transitions to GameChosen', async () => {
+    const initialState = gameState(localStatesA.lobby);
+    const action = joinOpenGame(bName, bAddress, stake);
+    const state = channelStates.preFund0;
+
+    const { storeState } = await expectSaga(gameSaga as any, client)
+      .withReducer(reducer, initialState)
+      .dispatch(action)
+      .provide([callCreateChannel(state)])
```

Comment:
It looks like `callCreateChannel(state)` returns an array. Do you need to put that array in another array here?

Reply:
Yeah. You can `provide` many stubs at once, and each one is an array, for some reason.